service/book/book_service.go

```
@@ -28,14 +29,44 @@ func (s svc) GetFilters(ctx *krogo.Context, filter string) ([]string, error) {
 	return s.store.GetFilters(ctx, filter)
 }
 
-func (s svc) Create(ctx *krogo.Context, book *model.Book, user *model.User) (*model.Book, error) {
-	return nil, nil
+func (s svc) Create(ctx *krogo.Context, book *model.Book, user *model.User) (*model.BookRes, error) {
+	if book == nil {
+		return nil, errors.InvalidParam{Param: []string{"invalid body request"}}
+	}
+
+	resp, err := s.store.Create(ctx, book)
+
+	if err != nil {
+		return nil, errors.DB{}
+	}
+
+	return resp, nil
 }
 
-func (s svc) Update(ctx *krogo.Context, book *model.Book, user *model.User) (*model.Book, error) {
-	return nil, nil
+func (s svc) Update(ctx *krogo.Context, book *model.Book, user *model.User) (*model.BookRes, error) {
+	if book == nil {
+		return nil, errors.InvalidParam{Param: []string{"invalid body request"}}
+	}
+
+	resp, err := s.store.Update(ctx, book)
+
+	if err != nil {
+		return nil, errors.DB{}
+	}
+
+	return resp, nil
 }
 
 func (s svc) Delete(ctx *krogo.Context, id uuid.UUID, user *model.User) error {
+	if id == uuid.Nil {
+		return errors.InvalidParam{Param: []string{"id"}}
+	}
+
+	err := s.store.Delete(ctx, id)
+
+	if err != nil {
+		return errors.DB{}
```

return the err received from store layer instead of an empty errors.DB{}